repositories/ingestion_repository.go

```diff
@@ -13,24 +13,24 @@ import (
 )
 
 type IngestionRepository interface {
-	IngestObjects(ctx context.Context, exec Executor, payloads []models.ClientObject, table models.Table) (int, error)
+	IngestObjects(ctx context.Context, tx Transaction, payloads []models.ClientObject, table models.Table) (int, error)
```

Comment:
This is something I wanted to do for a while: this repository method _expects_ a transaction as executor, to make sure the operations are done atomically. In other cases, it's up to the caller how he uses it.